### PR TITLE
BUGFIX: TMA-1571 Users Brick fails when segments_filter is used

### PR DESCRIPTION
**Pull request by hung-nguyen-hoang**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/17/ : SUCCESS in 7s

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 1]
```diff
@@ -262,7 +263,13 @@ def call(params)
                         fail "Client id cannot be empty" if client_id.blank?
 
                         c = domain_clients.detect { |specific_client| specific_client.id == client_id }
-                        fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\"" if c.nil?
+                        if c.nil?
+                          filtered_client = all_domain_clients.detect { |f_client| f_client.id == client_id }
+                          fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\"" if filtered_client.nil?
+
+                          GoodData.logger.info("Client #{client_id} is outside specific clients in segments_filter #{segment_uris}")
```
Comment (phong-nguyen-duy) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: please update the message:
"Client \"#{client_id}\" is not belong to filtered segments"

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/26/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/382/ : SUCCESS in 19s

**Pull request merged**